File: Cap 10/Topico 10.7/calcfat/calcfat3.js
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.7
 * 
 * Programa ..: OPERACAO COM SUBROTINA DE FUNCAO
 */

const readline = require('readline-sync');

function FATORIAL(N) {
  let I;
  let FAT = 1;
  for (I = 1; I <= N; I++) {
    FAT = FAT * I;
  }
  return FAT;
}

let LIMITE;
let RESP;
console.log("CALCULO DE FATORIAL");
console.log("FUNCAO SIMPLES SEM PASSAGEM DE PARAMETRO\n");
LIMITE = parseInt(readline.question("Qual fatorial (1-21): "));
console.log();
RESP = FATORIAL(LIMITE);
console.log("Fatorial = " + RESP);

File: Cap 10/Topico 10.7/calcfat/calcfat3.scala
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.7
 * 
 * Programa ..: OPERACAO COM SUBROTINA DE FUNCAO
 * 
 */

object Main {

  def fatorial(N: Byte): Long = {
    var FAT: Long = 1
    for (I <- 1 to N by 1) {
      FAT = FAT * I
    }
    FAT
  }

  var LIMITE: Byte = 0
  var RESP: Long = 0

  def main(args: Array[String]): Unit = {
  
    println("CALCULO DE FATORIAL")
    println("FUNCAO SIMPLES SEM PASSAGEM DE PARAMETRO")
    println()
	
    print("Qual fatorial (1-20): ")
    val input = scala.io.StdIn.readLine()
    LIMITE = input.toByte
	
    RESP = fatorial(LIMITE)
    println(s"Fatorial = $RESP")
	
  }
  
}
